File: leibniz.cpp
#include <iostream>
#include <time.h>
#include <string>

int main() {
    double pi = 4.0;
    int sign = -1;
    clock_t start_at = clock();
    for (int i = 3; i < 2147483647; i += 2) {
        pi = pi + (double) sign * 4 / i;
        sign *= -1;
    }
    std::cout << std::to_string(pi) << " consume " << (int) (clock() - start_at) / 1000 << " msec" << std::endl;
    return 0;
}

File: leibniz.java
public class leibniz {
    public static void main(String[] args) {
        float pi = 4.0f;
        int sign = -1;
        long start_at = System.currentTimeMillis();
        for (int i = 3; i < 2147483647; i += 2) {
            pi = pi + (float) sign * 4 / i;
            sign *= -1;
        }
        System.out.println(pi + " consume " + (System.currentTimeMillis() - start_at) + " msec");
    }
}
